exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,201 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+public class DominoesTest {
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none(); 
+	
+	@Test
+	public void emtpyInputEmptyOutputTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		ArrayList<Domino> dominoesArray = new ArrayList<>();
+		
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("Empty input domino list.");
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
```

Comment:
You don't have to assign to a variable here  (and in the other tests that expect an exception) since it's never used 🙂

And from looking at the canonical data it doesn't look like this should throw an error, it should just return an empty list I think?